### PR TITLE
[SPARK-31954][SQL] Delete duplicate testcase in HiveQuerySuite

### PR DESCRIPTION
**Pull request by GuoPhilipse**
### What changes were proposed in this pull request?
remove duplicate test cases

### Why are the changes needed?
improve test quality

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
No  test

### CODE REVIEW

**Comment by maropu** [comment]
ok to test

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala [hunk 1]
```diff
@@ -564,9 +564,6 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
     assert(1200 == res.getInt(0))
   }
 
-  createQueryTest("timestamp cast #4",
-    "SELECT CAST(CAST(1.2 AS TIMESTAMP) AS DOUBLE) FROM src LIMIT 1")
```
Comment (maropu) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala:568]: hm, as you suggested, the test looks duplicate. cc: @HyukjinKwon @viirya

**Review by HyukjinKwon — APPROVED** [review]
Looks good if tests pass

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala [hunk 2]
```diff
@@ -580,9 +577,6 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
     assert(-1200 == res.getInt(0))
   }
 
-  createQueryTest("timestamp cast #8",
-    "SELECT CAST(CAST(-1.2 AS TIMESTAMP) AS DOUBLE) FROM src LIMIT 1")
-
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala:585]: Yeah, we can rename them to make it prettier.

[the remark this answers is not in the document]
Comment (maropu) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala:585]: +1

**Comment by GuoPhilipse** [comment]
have adjusted test case order and rename test result file to look more pretty

**Comment by GuoPhilipse** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.0 and branch-2.4.

**Pull request closed without merging**